app/questionnaire/router.py

```diff
@@ -82,22 +91,26 @@ def can_access_location(self, location: Location, routing_path):
 
         return location.block_id in self._get_allowable_path(routing_path)
 
-    def can_access_hub(self):
+    def can_access_hub(self) -> bool:
         return self._schema.is_flow_hub and all(
             self._progress_store.is_section_complete(section_id)
             for section_id in self._schema.get_section_ids_required_for_hub()
             if section_id in self.enabled_section_ids
         )
 
-    def can_display_section_summary(self, section_id, list_item_id=None):
+    def can_display_section_summary(
+        self, section_id: str, list_item_id: str = None
+    ) -> Union[ImmutableDict, None, bool]:
         return self._schema.get_summary_for_section(
             section_id
         ) and self._progress_store.is_section_complete(section_id, list_item_id)
```

Comment:
This should always return a bool?

Reply:
It doesn't always return `bool` because we don't have a fallback option in here:
<img width="676" alt="Screenshot 2021-11-18 at 09 03 39" src="https://user-images.githubusercontent.com/53475968/142384573-22aca1df-92ed-41b0-ae07-c0032690f741.png">
so it returns `None` sometimes.

Reply:
it would work for router if we did something like this:

<img width="610" alt="Screenshot 2021-11-18 at 09 16 07" src="https://user-images.githubusercontent.com/53475968/142386509-fb733af3-9c3e-4369-976b-452bb888de54.png">

... but that would break `questionnaire_schema.py` types in many places.

Reply:
This is because we are simply using the existence of that dict object to see if we can access the summary. So you just need to cast it to a `bool`. You can either cast to bool, or check the condition:

```py
    def can_display_section_summary(
        self, section_id: str, list_item_id: str = None
    ) -> bool:
        return bool(
            self._schema.get_summary_for_section(section_id)
        ) and self._progress_store.is_section_complete(section_id, list_item_id)
```

or

```py
    def can_display_section_summary(
        self, section_id: str, list_item_id: str = None
    ) -> bool:
        return self._schema.get_summary_for_section(
            section_id
        ) is not None and self._progress_store.is_section_complete(
            section_id, list_item_id
        )
```

Reply:
I used the first approach, thanks!